service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpSend.java

```diff
@@ -121,17 +105,19 @@ public void send(String deviceId, String moduleId, Message message) throws IOExc
                 {
                     // Codes_SRS_SERVICE_SDK_JAVA_AMQPSEND_28_006: [The function shall create a binary message with the given content with deviceId only if moduleId is null]
                     amqpSendHandler.createProtonMessage(deviceId, message);
+                    log.info("Sending cloud to device message");
                 }
                 else
                 {
                     // Codes_SRS_SERVICE_SDK_JAVA_AMQPSEND_28_001: [The function shall create a binary message with the given content with moduleId]
                     amqpSendHandler.createProtonMessage(deviceId, moduleId, message);
+                    log.info("Sending cloud to device module message");
                 }
-                // Codes_SRS_SERVICE_SDK_JAVA_AMQPSEND_28_002: [The function shall initialize the Proton reactor object]
-                this.reactor = Proton.reactor(this);
-                // Codes_SRS_SERVICE_SDK_JAVA_AMQPSEND_28_003: [The function shall start the Proton reactor object]
-                this.reactor.run();
-                this.reactor.free();
+
+                new ReactorRunner(amqpSendHandler, "AmqpSend").run();
+
```

Comment:
No verifyConnectionWasOpened here?

Reply:
It does call ```sendComplete()``` which calls verifyConnectionWasOpened, while also checking the amqp ack code that the sent message got. I've renamed ```sendComplete()``` to ```verifySendSucceeded()``` to make this more clear